### ISSUE TITLE
Unintuitive behavior with Bigarray.Genarray.init for 0d arrays

### ISSUE THREAD

**Issue by mt-caret** [ocaml/ocaml#11570]
Hello,

The following code results in `0.` instead of `5.` as the user might expect; the underlying reason is that when passing an empty array to `Bigarray.Genarray.init`, it does not call the passed function at all. Is changing the behavior here so that the passed function is called once with the argument `[||]` reasonable?

```ocaml
let x = Bigarray.(Genarray.init float64 c_layout [||] (fun _ -> 5.)) in Bigarray.Genarray.get x [||]
```

**Comment by xavierleroy** [comment]
Thank you for a very polite report.  This does look like a bug to me, and the behavior you suggest is definitely the right one.

**Comment by nojb** [comment]
Fix in #11576

**Comment by mt-caret** [comment]
Woah thank you both for the super-quick response!
